addons/website_blog/static/src/js/systray_items/new_content.js

```diff
@@ -6,7 +6,20 @@ patch(NewContentModal.prototype, {
         super.setup();
 
         const newBlogElement = this.state.newContentElements.find(element => element.moduleXmlId === 'base.module_website_blog');
-        newBlogElement.createNewContent = () => this.onAddContent('website_blog.blog_post_action_add', true);
+        const context =
+          window.location.pathname === "/blog"
+            ? undefined
+            : {
+                default_blog_id:
+                  parseInt(
+                    document
+                      .querySelector("iframe.o_iframe")
+                      .contentDocument.querySelector(
+                        "main #wrap.website_blog [data-oe-model='blog.blog']"
+                      )?.dataset.oeId
+                  ) || undefined,
+              };
+        newBlogElement.createNewContent = () => this.onAddContent('website_blog.blog_post_action_add', true, context);
```

Comment:
This here can be improved, make it more 
```suggestion
      const isBlogPage = window.location.pathname === "/blog";
      let context = undefined;

      if (!isBlogPage) {
          const iframe = document.querySelector("iframe.o_iframe");
          const blogElement = iframe?.contentDocument?.querySelector(
              "main #wrap.website_blog [data-oe-model='blog.blog']"
          );
          const blogId = blogElement ? parseInt(blogElement.dataset.oeId) : undefined;
          context = {
              default_blog_id: blogId,
          };
      }

      newBlogElement.createNewContent = () =>
          this.onAddContent("website_blog.blog_post_action_add", true, context);
```

Reply:
I think both works the same, what I thought of is to use as few variables as we can and how we can achieve solution this with less LOC.
IMO first one feels good to go.
What do you think?